### PR TITLE
Sign transaction flow redesign

### PR DESCRIPTION
**Pull request by MiroMargineanu**
### Issue/Feature
A new layout for the sign transaction screens, layered after the send flow screens in Wallet, and supporting layout for the guardian code verification as well.

### Reproduce
Issue exists on version `2.13.8` of sdk-dapp.

### Contains breaking changes
[] Yes
[x] No

### Updated CHANGELOG
[x] Yes
[] No

### Testing
[x] User testing
[] Unit tests

### CODE REVIEW

**Review by razvantomegea — APPROVED** [review]

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 1]
```diff
@@ -75,15 +83,16 @@ export const SignStepBody = ({
...
+  const { isNft, isEgld, isEsdt } = getIdentifierType(tokenId);
 
   // If the token has a nonce means that this is an NFT. Eg: TokenId=TOKEN-1hfr, nonce=123 => NFT id=TOKEN-1hfr-123
   const appendedNonce = nonce ? `-${nonce}` : '';
   const nftId = `${tokenId}${appendedNonce}`;
 
-  const { tokenDecimals, tokenAvatar } = useGetTokenDetails({
-    tokenId: nonce && nonce.length > 0 ? nftId : tokenId
-  });
+  const { tokenDecimals, tokenAvatar, tokenLabel, type, price } =
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:92]: pirce --> egldPrice

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 2]
```diff
@@ -95,94 +104,72 @@ export const SignStepBody = ({
...
-  const showProgressSteps = totalSteps > 1;
   const classes = useSignStepsClasses(scamReport);
 
   const token = isNft ? nftId : tokenId || egldLabel;
   const shownAmount = isNft ? amount : formattedAmount;
 
+  const isNFTorSFT = [
+    NftEnumType.NonFungibleESDT,
+    NftEnumType.SemiFungibleESDT
+  ].includes(type as NftEnumType);
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:116]: no need to cast

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 3]
```diff
@@ -95,94 +104,72 @@ export const SignStepBody = ({
...
+
+  const { price: EGLDPrice } = useGetEgldPrice();
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:118]: why price twice?

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 4]
```diff
@@ -95,94 +104,72 @@ export const SignStepBody = ({
...
+  let tokenPrice = 0;
+
+  if (isEgld && EGLDPrice) {
+    tokenPrice = EGLDPrice;
+  }
+
+  if (!isNFTorSFT && !isEgld && price) {
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:125]: this should change

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  tokenAvatar?: string;
+}
+
+export const ConfirmFee = ({
+  transaction,
+  tokenAvatar,
+  egldLabel
+}: FeePropsType) => {
+  const { price } = useGetEgldPrice();
+  const feePrice = price || 0;
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx:24]: can fee price be zero?

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  egldLabel: string;
+  tokenAvatar?: string;
+}
+
+export const ConfirmFee = ({
+  transaction,
+  tokenAvatar,
+  egldLabel
+}: FeePropsType) => {
+  const { price } = useGetEgldPrice();
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx:23]: do you get 2 values from useGetEgldPrice ?

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  type: string;
+  avatar?: string;
+}
+
+export const TokenAvatar = (props: TokenAvatarPropsType) => {
+  const { avatar, type } = props;
+
+  if (type === NftEnumType.NonFungibleESDT) {
+    return (
+      <div className={classNames(styles.tokenAvatar, styles.tokenAvatarNFT)}>
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx:21]: use conditinonal classNames
[styles.tokenAvatarNFT]: isNft

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import classNames from 'classnames';
+
+import MultiversXIcon from 'assets/icons/EGLD.svg';
+import { NftEnumType } from 'types/tokens.types';
+
+import styles from './tokenAvatarStyles.scss';
+
+export interface TokenAvatarPropsType {
+  type: string;
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx:12]: use typed

File: src/hooks/transactions/useGetTokenDetails.tsx [hunk 9]
```diff
@@ -44,14 +62,19 @@ export function useGetTokenDetails({
...
 
-  const { isEsdt, isNft } = getIdentifierType(tokenId);
-  const tokenEndpoint = isEsdt ? TOKENS_ENDPOINT : COLLECTIONS_ENDPOINT;
-  let tokenIdentifier = tokenId;
+  let tokenEndpoint = TOKENS_ENDPOINT;
 
-  if (isNft) {
-    const [firstPart, secondPart] = tokenId.split('-');
-    tokenIdentifier = `${firstPart}-${secondPart}`;
+  if (isNft && tokenParts.length === nftSftTokenParts) {
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetTokenDetails.tsx:72]: use existing function for this

File: src/types/transactions.types.ts [hunk 10]
```diff
@@ -184,8 +184,9 @@ export interface GuardianScreenType extends WithClassnameType {
   title?: ReactNode;
   signStepInnerClasses?: SignStepInnerClassesType;
   signedTransactions?: DeviceSignedTransactions;
-  setSignedTransactions?: React.Dispatch<
-    React.SetStateAction<DeviceSignedTransactions | undefined>
+  guardianFormDescription?: ReactNode;
```
Comment (arhtudormorar) [inline src/types/transactions.types.ts:187]: guardianDescription / guardianSubtitle ?

File: src/utils/operations/calculateFeeInFiat.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  const feeInFiat = `≈ ${getUsdValue({
+    amount,
+    usd: egldPriceInUsd,
+    decimals: DIGITS
+  })}`;
+
+  return feeInFiat;
+}
+
+export default calculateFeeInFiat;
```
Comment (arhtudormorar) [inline src/utils/operations/calculateFeeInFiat.ts:29]: no export default

File: src/utils/operations/calculateFeeInFiat.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
+import { DIGITS, DECIMALS } from 'constants/index';
+
+import { formatAmount } from './formatAmount';
+import { getUsdValue } from './getUsdValue';
+
+export function calculateFeeInFiat({
```
Comment (arhtudormorar) [inline src/utils/operations/calculateFeeInFiat.ts:6]: add tests if copied from sdk-form

File: src/types/transactions.types.ts (hunk 10 above)
Comment (MiroMargineanu) [reply src/types/transactions.types.ts:187]: This is too generic, the description is strictly for the guardian input screen. 👍

File: src/utils/operations/calculateFeeInFiat.ts (hunk 11 above)
Comment (MiroMargineanu) [reply src/utils/operations/calculateFeeInFiat.ts]: Removed. 👍

File: src/utils/operations/calculateFeeInFiat.ts (hunk 12 above)
Comment (MiroMargineanu) [reply src/utils/operations/calculateFeeInFiat.ts]: Added. 👍

File: src/hooks/transactions/useGetTokenDetails.tsx [hunk 13]
```diff
@@ -76,16 +99,17 @@ export function useGetTokenDetails({
...
+    : '';
 
   return {
     tokenDecimals: tokenDecimals,
     tokenLabel,
     type: selectedToken?.type,
     tokenAvatar,
-    assets,
+    assets: selectedToken?.assets,
+    price: selectedToken?.price,
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/hooks/transactions/useGetTokenDetails.tsx]: Changed. 👍

File: src/hooks/transactions/useGetTokenDetails.tsx (hunk 9 above)
Comment (MiroMargineanu) [reply src/hooks/transactions/useGetTokenDetails.tsx:67]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx (hunk 8 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx]: Added. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/NFTSFTPreview/NFTSFTPreview.tsx [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+import styles from './NFTSFTPreviewStyles.scss';
+
+export interface NFTSFTPreviewPropsType {
+  txType: NftEnumType;
+  tokenLabel: string;
+  tokenId: string;
+  tokenAvatar: string;
+}
+
+export const NFTSFTPreview = ({
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/NFTSFTPreview/NFTSFTPreview.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx (hunk 5 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx (hunk 3 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx (hunk 2 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Removed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+    <div className={styles.token}>
+      <TokenAvatar type={tokenType} avatar={tokenAvatar} />
+
+      <div className={styles.value}>
+        {amount} <TokenDetails.Label token={token} />
+      </div>
+    </div>
+
+    {tokenPrice === null ? (
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx:35]: Discussed. It is treated. 👍

File: src/hooks/transactions/useGetTokenDetails.tsx [hunk 16]
```diff
@@ -44,14 +57,15 @@ export function useGetTokenDetails({
...
-  let tokenIdentifier = tokenId;
+  const tokenParts = tokenId.split('-');
+  const tokenIdentifier = tokenId;
 
-  if (isNft) {
-    const [firstPart, secondPart] = tokenId.split('-');
-    tokenIdentifier = `${firstPart}-${secondPart}`;
+  let tokenEndpoint = TOKENS_ENDPOINT;
+
+  if (isNft && tokenParts.length === nftSftTokenParts) {
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/hooks/transactions/useGetTokenDetails.tsx]: Changed. 👍

File: src/hooks/transactions/useGetTokenDetails.tsx [hunk 17]
```diff
@@ -1,11 +1,13 @@
 import axios from 'axios';
 import useSwr from 'swr';
 
-import { COLLECTIONS_ENDPOINT, TOKENS_ENDPOINT } from 'apiCalls/endpoints';
+import { NFTS_ENDPOINT, TOKENS_ENDPOINT } from 'apiCalls/endpoints';
 import { useGetNetworkConfig } from 'hooks/useGetNetworkConfig';
 import { NftEnumType } from 'types/tokens.types';
 import { getIdentifierType } from 'utils/validation/getIdentifierType';
 
+const nftSftTokenParts = 3;
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/hooks/transactions/useGetTokenDetails.tsx]: Removed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+    data: transaction.getData().toString(),
+    chainId: transaction.getChainID().valueOf()
+  });
+
+  return (
+    <div className={styles.fee}>
+      <span className={styles.label}>Fee</span>
+
+      <div className={styles.token}>
+        <TokenAvatar type='EGLD' avatar={tokenAvatar} />
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 19]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
+          />
+
+          <div className={styles.columns}>
+            {shouldShowAmount && (
+              <div className={styles.column}>
+                <ConfirmAmount
+                  tokenAvatar={tokenAvatar}
+                  amount={shownAmount}
+                  token={token}
+                  tokenType={isEgld ? 'EGLD' : type ?? undefined}
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx (hunk 7 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/TokenAvatar/TokenAvatar.tsx]: Changed. 👍

File: src/UI/LoadingDots/LoadingDots.tsx [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+export const LoadingDots = ({ className }: WithClassnameType) => (
+  <div className={classNames(styles.loadingDots, className)}>
+    {Array.from({ length: 3 }).map((_, index) => (
+      <span
+        key={`loading-dot-${index}`}
+        className={classNames(styles.loadingDot, styles[`loadingDot-${index}`])}
+      />
+    ))}
+
+    <span>Loading...</span>
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/LoadingDots/LoadingDots.tsx:17]: Not possible, we need something to simulate existing text, passing an empty string isn't taken into account. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 21]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
+
+  if (isEgld && egldPrice) {
+    tokenPrice = egldPrice;
+  }
+
+  if (isNft) {
+    tokenPrice = 0;
+  }
+
+  if (isEsdt && type) {
```
[the remark this answers is not in the document]
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:127]: It also exists for these, it comes as `FungibleESDT`. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 22]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
+    tokenPrice = esdtPrice ?? 0;
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:128]: do not use 0

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 23]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
+  const isNftOrSft = type === NftEnumType.SemiFungibleESDT || isSft;
+
+  const { price: egldPrice } = useGetEgldPrice();
+  let tokenPrice = null;
+
+  if (isEgld && egldPrice) {
+    tokenPrice = egldPrice;
+  }
+
+  if (isNft) {
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:123]: to be removed

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 24]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
-  const totalSteps = allTransactions.length + extraGuardianStep;
   const scamReport = currentTransaction.receiverScamInfo;
-  const showProgressSteps = totalSteps > 1;
   const classes = useSignStepsClasses(scamReport);
 
   const token = isNft ? nftId : tokenId || egldLabel;
   const shownAmount = isNft ? amount : formattedAmount;
 
+  const isSft = type === NftEnumType.SemiFungibleESDT;
+  const isNftOrSft = type === NftEnumType.SemiFungibleESDT || isSft;
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:114]: to be removed

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx [hunk 25]
```diff
@@ -95,94 +104,74 @@ export const SignStepBody = ({
...
+          />
+
+          <div className={styles.columns}>
+            {shouldShowAmount && (
+              <div className={styles.column}>
+                <ConfirmAmount
+                  tokenAvatar={tokenAvatar}
+                  amount={shownAmount}
+                  token={token}
+                  tokenType={isEgld ? egldLabel : type ?? undefined}
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx:163]: no need for undefined

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+      <LoadingDots className={styles.price} />
+    ) : tokenPrice === 0 ? null : (
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx:37]: maybe no need to check for 0

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+
+export const ConfirmFee = ({
+  transaction,
+  tokenAvatar,
+  egldLabel
+}: FeePropsType) => {
+  const { price } = useGetEgldPrice();
+
+  const feeLimit = calculateFeeLimit({
+    gasPerDataByte: String(GAS_PER_DATA_BYTE),
```
Comment (arhtudormorar) [inline src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx:27]: include isGuarded

File: src/hooks/transactions/useGetTokenDetails.tsx [hunk 28]
```diff
@@ -44,14 +55,13 @@ export function useGetTokenDetails({
   tokenId: string;
 }): TokenOptionType {
   const { network } = useGetNetworkConfig();
+  const { isNft } = getIdentifierType(tokenId);
 
-  const { isEsdt, isNft } = getIdentifierType(tokenId);
-  const tokenEndpoint = isEsdt ? TOKENS_ENDPOINT : COLLECTIONS_ENDPOINT;
-  let tokenIdentifier = tokenId;
+  const tokenIdentifier = tokenId;
+  let tokenEndpoint = TOKENS_ENDPOINT;
```
Comment (arhtudormorar) [inline src/hooks/transactions/useGetTokenDetails.tsx:61]: use ternary

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx (hunk 22 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Changed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx (hunk 24 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Removed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx (hunk 25 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/SignStepBody.tsx]: Removed. 👍

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx (hunk 26 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx]: Changed. 👍

File: src/hooks/transactions/useGetTokenDetails.tsx (hunk 28 above)
Comment (MiroMargineanu) [reply src/hooks/transactions/useGetTokenDetails.tsx]: Changed. 👍

**Review by arhtudormorar — APPROVED** [review]

File: src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx (hunk 27 above)
Comment (MiroMargineanu) [reply src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmFee/ConfirmFee.tsx:27]: Not necessary anymore. Discussed. 👍

**Pull request merged**